mysql_mimic/variables_processor.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime
+
+from sqlglot import expressions as exp
+
+from mysql_mimic.intercept import value_to_expression
+from mysql_mimic.variables import Variables
+
+
+class SessionContext:
```

Comment:
Can this just be a dataclass?

Reply:
Or perhaps this should be a Protocol? And we do `VariablesProcessor(self)` in the session? Or maybe that's getting too fancy.